kinto/plugins/history/views.py

```diff
@@ -43,3 +45,88 @@ def _extract_filters(self, queryparams=None):
             filters_str_id.append(filt)
 
         return filters_str_id
+
+
+class Revert(resource.UserResource):
+
+    def __init__(self, request):
+        self.request = request
+        self.bucket_id = request.matchdict['bucket_id']
+        if 'since' not in request.params:
+            raise_invalid(self.request, 'params', 'since', 'since request parameter must be provided')
+        self.since = int(request.params['since'])
+        filters = super(Revert, self)._extract_filters(request.params)
+        filters = [s for s in filters
+                                 if s.field != 'since']
+        self.storage = request.registry.storage
+        self.history, _ = self.storage.get_all(parent_id='/buckets/%s' % self.bucket_id,
+                                               collection_id='history', filters=filters,
+                                               sorting=[Sort('last_modified', -1)])
+
+    def batch_revert(self, revert_list):
+        batch = {'path': '/batch', 'method': 'POST', 'body': {'requests': revert_list}}
+        subrequest = build_request(self.request, batch)
+        try:
+            resp, subrequest = self.request.follow_subrequest(subrequest,
+                                                         use_tweens=False)
+        except httpexceptions.HTTPException as e:
+            if e.content_type == 'application/json':
+                resp = e
+            else:
+                resp = errors.http_error(e)
+        return build_response(resp, subrequest)
+
+    def revert_changes(self):
+        revert_list = self.build_revert_list()
+        if len(revert_list) == 0:
+            raise_invalid(self.request, 'params', 'no history', 'there are no history records to revert after since timestamp')
```

Comment:
Maybe this is not a problem after all. I didn't see any test for this for example :)